src/searching/binary_search_recursive.rs

```diff
@@ -0,0 +1,226 @@
+use std::cmp::Ordering;
+/// ## Binary Search algorithm
+/// Iterating over the array by dividing half of the array and returns the index value of the target element in the array.
```

Comment:
Maybe use a different wording since this is the recursive variant? That is, don't use "Iterating" unless it is explicitly an iteration.